webapp/plugins/pluggable.jsx

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class Pluggable extends React.PureComponent {
+    static propTypes = {
+        children: PropTypes.element.isRequired
+    }
+
+    render() {
+        const child = this.props.children.type;
+
+        if (child == null) {
+            return null;
+        }
+
+        // Include any props passed to this component or to the child component
+        const props = {...this.props, ...this.props.children.props};
+        Reflect.deleteProperty(props, 'children');
```

Comment:
This will delete the children of the child component if we ever want something like
```
<Pluggable>
    <Foo>
         <Bar/>
         <Bar/>
    </Foo>
</Pluggable>
```